cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -153,45 +146,53 @@ void contingencyMatrix(const T* groundTruth,
  *        labels. Users should call function getInputClassCardinality to find
  *        and allocate memory for output. Similarly workspace requirements
  *        should be checked using function getContingencyMatrixWorkspaceSize
- * @tparam T label type
- * @tparam OutT output matrix type
+ * @tparam DataT label type
+ * @tparam OutType output matrix type
  * @tparam IdxType Index type of matrix extent.
  * @tparam LayoutPolicy Layout type of the input data.
- * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
- *                        device.
  * @param handle: the raft handle.
  * @param groundTruth: device 1-d array for ground truth (num of rows)
  * @param predictedLabel: device 1-d array for prediction (num of columns)
- * @param outMat: output buffer for contingecy matrix
+ * @param outMat: output buffer for contingency matrix
  * @param workspace: Optional, workspace memory allocation
- * @param workspaceSize: Optional, size of workspace memory
  * @param minLabel: Optional, min value in input ground truth array
  * @param maxLabel: Optional, max value in input ground truth array
  */
-template <typename T,
-          typename OutT = int,
+template <typename DataT,
+          typename OutType,
           typename IdxType,
           typename LayoutPolicy,
-          typename AccessorPolicy>
-void contingencyMatrix(
-  const raft::handle_t& handle,
-  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> groundTruth,
-  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> predictedLabel,
-  raft::mdspan<OutT, raft::matrix_extent<IdxType>, LayoutPolicy, AccessorPolicy> outMat,
-  void* workspace      = nullptr,
-  size_t workspaceSize = 0,
-  T minLabel           = std::numeric_limits<T>::max(),
-  T maxLabel           = std::numeric_limits<T>::max())
+          typename WorkspaceType,
```

Comment:
Given that the code assumes `WorkspaceType` is some kind of mdspan, would you consider just spelling that out in its type?  `workspace` would just be `std::optional<raft::device_vector_view<DataT, IdxType>>`.  That way, you wouldn't even need a template parameter.

If you want to say that `WorkspaceType` could be any device-accessible mdspan, you could use `mdspan<DataT, extents<WorkspaceIndexType, WorkspaceExtent>, layout_right, Accessor>`.  For rank-1 mdspan, `layout_left` can be converted to `layout_right` and vice versa.  Then, just constrain the Accessor to be device accessible, using RAFT's traits.